master/internal/api_experiment.go

```diff
@@ -43,3 +53,104 @@ func (a *apiServer) GetExperiments(
 }
+
+func (a *apiServer) ActivateExperiment(
+	ctx context.Context, req *apiv1.ActivateExperimentRequest) (
+	resp *apiv1.ActivateExperimentResponse, err error) {
+	var rawExp []byte
+	var exp model.Experiment
+	if rawExp, err = a.m.db.ExperimentRaw(int(req.Id)); err != nil {
+		return nil, status.Errorf(codes.NotFound, "%s; experiment %d not found.", err.Error(), req.Id)
+	}
+	if err = json.Unmarshal(rawExp, &exp); err != nil {
+		return nil, status.Error(codes.Internal, "failed to unmarshal experiment")
+	}
+
+	addr := actor.Addr("experiments", req.Id).String()
+	rCode, err := a.actorRequestExpanded(addr, req, &resp)
+	if rCode == codes.NotFound {
+		return nil, badStateErr
+	}
+	return resp, err
+}
+
+func (a *apiServer) PauseExperiment(
+	ctx context.Context, req *apiv1.PauseExperimentRequest) (
+	resp *apiv1.PauseExperimentResponse, err error) {
+	var rawExp []byte
+	var exp model.Experiment
+	if rawExp, err = a.m.db.ExperimentRaw(int(req.Id)); err != nil {
+		return nil, status.Errorf(codes.NotFound, "%s; experiment %d not found.", err.Error(), req.Id)
+	}
+	if err = json.Unmarshal(rawExp, &exp); err != nil {
+		return nil, status.Error(codes.Internal, "failed to unmarshal experiment")
+	}
+
+	addr := actor.Addr("experiments", req.Id).String()
+	rCode, err := a.actorRequestExpanded(addr, req, &resp)
+	if rCode == codes.NotFound {
+		return nil, badStateErr
+	}
+	return resp, err
+}
+
+func (a *apiServer) CancelExperiment(
+	ctx context.Context, req *apiv1.CancelExperimentRequest) (
+	resp *apiv1.CancelExperimentResponse, err error) {
+	var rawExp []byte
+	var exp model.Experiment
+	if rawExp, err = a.m.db.ExperimentRaw(int(req.Id)); err != nil {
+		return nil, status.Errorf(codes.NotFound, "%s; experiment %d not found.", err.Error(), req.Id)
+	}
+	if err = json.Unmarshal(rawExp, &exp); err != nil {
+		return nil, status.Error(codes.Internal, "failed to unmarshal experiment")
+	}
+
+	addr := actor.Addr("experiments", req.Id).String()
+	rCode, err := a.actorRequestExpanded(addr, req, &resp)
+	if rCode == codes.NotFound {
+		return &apiv1.CancelExperimentResponse{}, nil
+	}
+	return resp, err
+}
+
+func (a *apiServer) KillExperiment(
+	ctx context.Context, req *apiv1.KillExperimentRequest) (
+	resp *apiv1.KillExperimentResponse, err error) {
+	var rawExp []byte
+	var exp model.Experiment
+	if rawExp, err = a.m.db.ExperimentRaw(int(req.Id)); err != nil {
+		return nil, status.Errorf(codes.NotFound, "%s; experiment %d not found.", err.Error(), req.Id)
+	}
+	if err = json.Unmarshal(rawExp, &exp); err != nil {
+		return nil, status.Error(codes.Internal, "failed to unmarshal experiment")
+	}
+
+	addr := actor.Addr("experiments", req.Id).String()
+	rCode, err := a.actorRequestExpanded(addr, req, &resp)
+	if rCode == codes.NotFound {
+		return &apiv1.KillExperimentResponse{}, nil
+	}
+	return resp, err
+}
+
+func (a *apiServer) setArchiveStatus(reqID int32, doArchive bool) (
```

Comment:
blocking: who is using the first parameter?

Reply:
are you asking if it's used? https://github.com/determined-ai/determined/pull/719/files/60cf271a5a0bdf676f9fdf7a059f047fa9773799#diff-27c0ef1764855c633d9729f443451bddR140
https://github.com/determined-ai/determined/pull/719/files/60cf271a5a0bdf676f9fdf7a059f047fa9773799#diff-27c0ef1764855c633d9729f443451bddR147

Reply:
`ArchiveExperiment(` and Unarchive

Reply:
The first parameter is ignored in both calls below.

Reply:
oh you mean the return value? `*experimentv1.Experiment,` ?